### PR TITLE
[OSX] Set MACOSX_DEPLOYMENT_TARGET to remove linking warning

### PR DESCRIPTION
**Pull request by mtojek**
This PR tries to git rid of warning:

```
ld: warning: object file (/var/folders/dt/lz9gptg91vzchq7zp5grt85h0000gn/T/go-link-168878093/000012.o) was built for newer OSX version (10.15) than being linked (10.13)
```
Sample build output with warning: https://travis-ci.org/elastic/beats/jobs/627582568?utm_medium=notification&utm_source=github_status

Issue: https://github.com/elastic/beats/issues/15203

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ env:
     - TRAVIS_GO_VERSION=$(cat .go-version)
     # Newer versions of minikube fail on travis, see: https://github.com/kubernetes/minikube/issues/2704
     - TRAVIS_MINIKUBE_VERSION=v0.25.2
+    - MACOSX_DEPLOYMENT_TARGET=10.15
```
Comment (jsoriano) [inline .travis.yml]: Do you know why is this needed?

Comment (mtojek) [reply .travis.yml]: There is a linking issue in the newer Go releases. See similar topics: https://github.com/ldc-developers/ldc/issues/1291
